Plugins/Editor/Scripts/Control/Helpers/CracksStitching.cs

```diff
@@ -0,0 +1,514 @@
+                        }
+
+                        // Both edges are each other's best match
+                        leftToSolve.Remove(thisEdge);
+                        leftToSolve.Remove(bestMatch.edge);
+                        
+                        int index = trianglesToAdd.Count;
+                        CreateTriangles(ref workingData, ref thisEdge, ref bestMatch.edge);
+                        debug?.Log($"{thisEdge} -> {bestMatch.edge}: {trianglesToAdd[index]} {(trianglesToAdd.Count - index > 1 ? trianglesToAdd[index+1].ToString() : "")}");
+                        #if YIELD_SUBSTEPS
+                        yield return null;
+                        #endif
+                        break;
+                    }
+                    else
+                    {
+                        // They don't match, try to see if this new best match matches each other instead
+                        thisEdge = bestMatch.edge;
+                        bestMatch = otherBestMatch;
+                        swapCount++;
+                        #if YIELD_SUBSTEPS
+                        yield return null;
+                        #endif
+                        continue;
+                    }
+                } while (true);
+            }
+
+            var posToSubMesh = new Dictionary<Vector3, (int submesh, int vertIndex)>();
+            for (int subMeshIndex = 0; subMeshIndex < subMeshes.Length; subMeshIndex++)
+            {
+                var subMesh = subMeshes[subMeshIndex];
+                for (int i = 0; i < subMesh.Count; i++)
+                {
+                    var subMeshVertIndex = subMesh[i];
+                    var pos = nativeVertices[subMeshVertIndex];
+                    // Multiple assignment on the same key would matter
+                    // only for large holes next to multiple materials,
+                    // we do not expect cracks to be large enough to warrant solving for this.
+                    posToSubMesh[pos] = (subMeshIndex, subMeshVertIndex);
+                }
+            }
+
+            foreach (var (x, y, z) in trianglesToAdd)
+            {
+                var (subMeshIndex, mappingA) = posToSubMesh[vertices[x]];
+                var (           _, mappingB) = posToSubMesh[vertices[y]];
+                var (           _, mappingC) = posToSubMesh[vertices[z]];
+                
+                // Effectively randomly picking subMesh, i.e.: material, those triangles will be assigned to, see comment above
+                var indices = subMeshes[subMeshIndex];
+                
+                // Not sure yet how to properly solve winding order, add both sides for now
+                indices.Add(mappingA);
+                indices.Add(mappingB);
+                indices.Add(mappingC);
+                indices.Add(mappingC);
+                indices.Add(mappingB);
+                indices.Add(mappingA);
+            }
```

Comment:
Small issue right now is the fact that I'm not yet sure how to solve triangle winding order, so I'm adding triangles facing both sides. If you have a suggestion I am open to implement it.